src/ClassicUO.IO/ReaderUtil.cs

```diff
@@ -0,0 +1,24 @@
+﻿// SPDX-License-Identifier: BSD-2-Clause
+// author: Max Kellermann <max.kellermann@gmail.com>
+
+using System;
+using System.Text;
+
+namespace ClassicUO.IO
+{
+    /// <summary>
+    ///     Utilities for reading data from raw buffers.
+    /// </summary>
+    public class ReaderUtil
+    {
+        public static string ReadFixedSizeString(ReadOnlySpan<byte> src)
+        {
+            return Encoding.UTF8.GetString(MemoryExtensions.TrimEnd(src, (byte)0));
```

Comment:
nit, just FYI - this is an extension method, so you could use this syntax too:
```suggestion
            return Encoding.UTF8.GetString(src.TrimEnd((byte)0));
```

Reply:
With that said, if you want to (doesn't matter to me), you can also make `ReadFixedSizeString` an extension method and reduce the usage to something like `landTilesPtr->Name.ReadFixedSizeString(20)`

Reply:
Hm, I tried calling `src.TrimEnd()` but got a compiler error because it was only implemented for `<char>`. Now it works. I wonder why.

Reply:
At the time you probably had not casted `0` to a `byte` yet, and the generic type must match to fulfill the extension method signature. It's a common issue.